--- ./lib/copy.h	2015-05-01 22:00:49.170294000 -0700
+++ ./lib/timespec.h	2015-05-01 22:04:17.240216000 -0700
@@ -45,8 +45,13 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  signed long long int diff = (signed long long int)a.tv_sec - (signed long long int)b.tv_sec;
+  if (diff > 0)
+    return 1;
+  if (diff < 0)
+    return -1;
+  if (diff == 0)
+    return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
